app/helpers/jobs_helper.rb

```diff
@@ -1,2 +1,10 @@
 module JobsHelper
+  def filter_jobs(jobs, filter)
+    @right_jobs = jobs
+    @right_jobs = @right_jobs.where(:title => filter[:title]) if filter[:title].present?
+    @right_jobs = @right_jobs.where(:location => filter[:location]) if filter[:location].present?
+    @right_jobs = @right_jobs.where(:company => filter[:company]) if filter[:company].present?
+    @right_jobs = @right_jobs.where(:category => filter[:category]) if filter[:category].present?
+    @jobs = @right_jobs
+  end
```

Comment:
Why do you expect any ```@jobs``` here? what is it's not defined? you can do it simple like : 
```Rails
def filter_jobs(jobs, filter)
  jobs = jobs.where(:title => filter[:title]) if filter[:title].present?
  jobs = jobs.where(:location => filter[:location]) if filter[:location].present?
  jobs = jobs.where(:company => filter[:company]) if filter[:company].present?
  jobs = jobs.where(:category => filter[:category]) if filter[:category].present?
  jobs
end
```